### ISSUE TITLE
Fix incorrect locations in no-extra-semicolons

### ISSUE THREAD

**Issue by andrea-guarino-sonarsource** [stylelint/stylelint#5153]
> Clearly describe the bug

Hello,
it looks like issues coming from `no-extra-semicolons` rule are reported on the wrong line when a non-css file (e.g. html) is analyzed.

> Which rule, if any, is the bug related to?

`no-extra-semicolons`

> What code is needed to reproduce the bug?

```html
<!doctype html>
<html lang="en">
<head>
  <title>Index</title>
  <style>
    .category-title { /* Issue raised here. */
    }  
    .category-filters {
      display: none;
    }
    .product-listing {
      justify-content: center;; /* Issue should have been raised here. */
    }
</style>
</head>
<body>
  <?= "Hello World!" ?>
</body>
</html>
```

Note that, in case of css file, issue is not reproducible:

```css
.category-title {
}
.category-filters {
  display: none;
}
.product-listing {
  justify-content: center;; /* Issue here */
}
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
    "extends": "stylelint-config-standard",
}
```

> Which version of stylelint are you using?

`13.10.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

I reproduced with CLI, Node.js API and vscode plugin.

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

No.

> What did you expect to happen?

Issue raised on the correct line (like for .css files).

> What actually happened (e.g. what warnings or errors did you get)?

Issue raised on the wrong line.

**Comment by jeddy3** [comment]
@andrea-guarino-sonarsource Thanks for the report and for using the template.

I can reproduce this [using the demo](https://stylelint.io/demo#N4Igxg9gJgpiBcIA8BCKEwBcCeAHGABABaYC2ANgHwA6AdkiRQeQIa0DmAvNSDLTzXpEYLKIIIEkmAJaZyMSgElasAB5IA9DLkK6EpAGcc88RIIA6MC0wx2EAE7YAtNvkFgBDQCoCigwYBXQnsWaQMYKGIYexhzAi8NPTMAXwkkiUtrWwdnADNpcht7A3d0sygw3FZseAJaCFoYAG4y5LLzXHtoAKwncjCZDlLaMzMAKwCjaVznSFobedqwPiKmps8fP0DCAyIIAPJIohYAN0IAIxg+AhCwiKiYuITWuk0jbBNXjWFRQSRz6DYUxIAD8nAIPAAEjByOQIAQAOoOQ4oHgEEF-DQAqBAr6MKggAA0IAM2HmLFUCBA+KJ4Aa+XYVOASR49gOMAMPFqzJGZh49ScMFUmBCTnCpGkkDhtE5CAIIqCSTatGSIGSQA).

I've labelled the issue as ready to implement. Please consider [contributing to the HTML parser](https://stylelint.io/developer-guide/syntaxes#working-on-syntaxes) if you have time.

**Comment by jeddy3** [comment]
Closing [as stylistic rules are frozen](https://github.com/stylelint/stylelint/issues/5674). The community is welcome to migrate the rule [to a plugin](https://stylelint.io/developer-guide/plugins) and fix this bug.
